internal/transform/trade.go

```diff
@@ -0,0 +1,210 @@
+		}
+
+		outputCounterAccountAddress, err := utils.GetAccountAddressFromMuxedAccount(sourceAccount)
+		if err != nil {
+			return []TradeOutput{}, err
+		}
+
+		var outputCounterAssetType, outputCounterAssetCode, outputCounterAssetIssuer string
+		err = claimOffer.AssetBought.Extract(&outputCounterAssetType, &outputCounterAssetCode, &outputCounterAssetIssuer)
+		if err != nil {
+			return []TradeOutput{}, err
+		}
+
+		outputCounterAmount := int64(claimOffer.AmountBought)
+		if outputCounterAmount < 0 {
+			return []TradeOutput{}, fmt.Errorf("Amount bought is negative (%d) for operation at index %d", outputCounterAmount, operationIndex)
+		}
+
+		if outputBaseAmount == 0 && outputCounterAmount == 0 {
+			return []TradeOutput{}, fmt.Errorf("Both base and counter amount are 0 for operation at index %d", operationIndex)
+		}
+
+		// Final price should be buy / sell
+		outputPriceN, outputPriceD := outputCounterAmount, outputBaseAmount
+
+		if err != nil {
+			return []TradeOutput{}, err
+		}
+
+		outputBaseIsSeller := true
+
+		trade := TradeOutput{
+			Order:                 outputOrder,
+			LedgerClosedAt:        outputLedgerClosedAt,
+			OfferID:               outputOfferID,
+			BaseAccountAddress:    outputBaseAccountAddress,
+			BaseAssetType:         outputBaseAssetType,
+			BaseAssetCode:         outputBaseAssetCode,
+			BaseAssetIssuer:       outputBaseAssetIssuer,
+			BaseAmount:            outputBaseAmount,
+			CounterAccountAddress: outputCounterAccountAddress,
+			CounterAssetType:      outputCounterAssetType,
+			CounterAssetCode:      outputCounterAssetCode,
+			CounterAssetIssuer:    outputCounterAssetIssuer,
+			CounterAmount:         outputCounterAmount,
+			BaseIsSeller:          outputBaseIsSeller,
+			PriceN:                outputPriceN,
+			PriceD:                outputPriceD,
+		}
+
+		transformedTrades = append(transformedTrades, trade)
+	}
+	return transformedTrades, nil
+}
+
+func extractClaimedOffers(operationResults []xdr.OperationResult, operationIndex int32, operationType xdr.OperationType) (claimedOffers []xdr.ClaimOfferAtom, err error) {
+	if operationIndex >= int32(len(operationResults)) {
+		err = fmt.Errorf("Operation index of %d is out of bounds in result slice (len = %d)", operationIndex, len(operationResults))
+		return
+	}
+
+	if operationResults[operationIndex].Tr == nil {
+		err = fmt.Errorf("Could not get result Tr for operation at index %d", operationIndex)
+		return
+	}
+
+	operationTr, ok := operationResults[operationIndex].GetTr()
+	if !ok {
+		err = fmt.Errorf("Could not get result Tr for operation at index %d", operationIndex)
+		return
+	}
+
+	switch operationType {
+	case xdr.OperationTypeManageBuyOffer:
+		offerResult, ok := operationTr.GetManageBuyOfferResult()
+		if !ok {
+			err = fmt.Errorf("Could not get %sResult for operation at index %d", operationType, operationIndex)
```

Comment:
I would make this error message explicit to `ManageBuyOfferResult`. While it would functionally be the same, but there's minimal space saving with the format string, and you'd make the developer's life a bit easier (since there are multiple branches with the same error message). Ultimately, error messages should be maximally readable and navigable, so the developer can easily find the source.